### DISCUSSION TITLE
Development Update, Re: New config management coming soon...

### DISCUSSION THREAD

**Discussion by lissy93** [Lissy93/dashy#799] [Announcements]
Just an update...

I'm currently working on re-doing how the config is loaded, managed and updated.
When this is done, it should make Dashy faster, easier to use, and more consistent. 

For reference, here are my requirements for version 2.1.2
- [ ] Config + assets mounted as a single directory
- [ ] Remove the need for a rebuild when config updated
- [ ] Add authentication to backend, so only Dashy can make requests to it
- [x] Data loaded dynamically on page load
- [ ] Make it clearer to the user what config being viewed / edited
- [ ] Improve security of backend endpoints
- [ ] Refactor schema into reusable sections
- [ ] Refactor main store to make is more readable
- [ ] Lay the groundwork for supporting an API
- [ ] Lay the groundwork for building UI editor for widgets

This should take about a week to complete, although life always has a way of getting in the way, so bear with me if it's a little over due.

And in the meantime, if you have a bug which directly relates to config loading, then please leave it as a comment below, and not a new ticket, as duplicates tickets are harder to keep track of.

Update, progress being tracked in [this branch](https://github.com/Lissy93/dashy/tree/REFACTOR/2.1.2_unified-config).

**Comment by RahmaNiftaliyev** [comment]
refactor schema

**Reply by lissy93** [reply]
It's on the list, number 7:  _"Refactor schema into reusable sections"_

**Comment by chevdor** [comment]
Awesome news, I was about to line up a PR along those lines.
Your 2 first points are a **huge** benefit :)
I think having all the config in a separate folder brings several benefits.

You could add to the list that the backups should be:
- opt-in: I personally use k8s and it is easy to script backups using cronjobs and also to save time when I edit the config
- in their separate folder that is user defined
- the backups should not be "watched" byt the file watcher that detects config changes
- the backups should be a zip of the whole config folder and not just the conf.yml

**Comment by GGORG0** [comment]
Just linking this issue as related: https://github.com/Lissy93/dashy/issues/877
Unfortunately, I didn't find this discussion before posting, sorry...

**Comment by lissy93** [comment]
Just an update:
Apologies this has been taking a while, it's been a busy Summer 
Got a 1 - 2 days work left, so hopefully will be merged within the next week or two.

**Reply by Tharic99** [reply]
> Just an update:
> Apologies this has been taking a while, it's been a busy Summer
> Got a 1 - 2 days work left, so hopefully will be merged within the next week or two.

Thank you for the update and all of your work with this, Lissy.

Are you expecting all of the items to be merged within the next few weeks or just the first two? I'm personally waiting on the first two to resolve some of my problems with updating the configuration inside of an Unraid docker container.

**Comment by chevdor** [comment]
Here is an issue that would be worth noticing regarding the config: https://github.com/Lissy93/dashy/issues/902

**Comment by Cereal916** [comment]
Thanks for the hard work Lissy. Love the app.

**Comment by jl-678** [comment]
@Lissy93 I think that Dashy is great!  Thank you.  Out of curiosity, do you have a new expected ETA on 2.1.2?  I am sure that you are swamped and so was hoping to get a bit more clarity on your plans. TIA!

**Comment by PichlAlex** [comment]
i want to implement a dashboard in my Lab ...and i need multiple pages.

however i have some problems while having a multiple page-configuration. When i navigate to a second page, the url changes from http://10.2.0.28:8080/ to http://10.2.0.28:8080/home/network2 ... from this on every request (including saving this page config) to dashy fails with 404 ... except navigating to http://10.2.0.28:8080/
i think this problem will be solved with issue https://github.com/Lissy93/dashy/issues/923

so i am really looking forward to get 2.1.2 finished :-)

Great idea Lizzy and good work Lizzy&contributors!

**Comment by svenniuwe** [comment]
Yeah, the first two points would be amazing.
Because in my case (installed it on Ubuntu 22.04 "bare metal" without Docker) I have to restart the service (the server) so that config updates become visible (in addition to re-building the app before).

**Reply by svenniuwe** [reply]
Update: Today it works sometimes just hitting Ctrl+F5. Nice :-)
